src/RewardsStreamerMP.sol

```diff
@@ -236,19 +258,35 @@ contract RewardsStreamerMP is ReentrancyGuard {
 
         bool success = REWARD_TOKEN.transfer(to, amount);
         if (!success) {
-            revert StakingManager__TransferFailed();
+            revert StakeManager__TransferFailed();
         }
     }
 
-    function getStakedBalance(address userAddress) external view returns (uint256) {
-        return users[userAddress].stakedBalance;
+    function calculateMP(uint256 _balance, uint256 _deltaTime) public pure returns (uint256) {
+        return (_deltaTime * _balance * MP_APY) / (365 days * SCALE_FACTOR);
+    }
+
+    function getStakedBalance(address _vault) external view returns (uint256 _balance) {
+        return users[_vault].stakedBalance;
+    }
+
+    function getPendingRewards(address _vault) external view returns (uint256) {
+        return calculateUserRewards(_vault);
+    }
+
+    function getUserInfo(address _vault) external view returns (UserInfo memory) {
+        return users[_vault];
+    }
+
+    function totalSupplyMinted() external view returns (uint256 _totalSupply) {
+        return totalStaked + totalMP;
     }
 
-    function getPendingRewards(address userAddress) external view returns (uint256) {
-        return calculateUserRewards(userAddress);
+    function totalSupply() external view returns (uint256 _totalSupply) {
+        return totalStaked + totalMP + potentialMP;
     }
 
-    function getUserInfo(address userAddress) external view returns (UserInfo memory) {
-        return users[userAddress];
+    function pendingReward() external view returns (uint256 _pendingReward) {
+        return STAKE_TOKEN().balanceOf(address(this)) - accountedRewards;
```

Comment:
I think the pending rewards are actually `potentialMP`. Otherwise if we only use `balanceOf` we lose the initial MPs

Reply:
I'll revist that.